packages/core/test/tools/create-cache.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const should = require('should');
+
+const { makeRpc, mockRpcCall } = require('./mocky');
+const createCache = require('../../src/tools/create-cache');
+
+describe('zcache: get, set, delete', () => {
+  const rpc = makeRpc();
+  const cache = createCache({ _zapier: { rpc } });
+
+  it('zcache_get: should return the cache entry of an existing key', async () => {
+    const value = {entity:'Zapier', colors: ['Orange', 'black']};
+    mockRpcCall(JSON.stringify(value));
+
+    const result = await cache.get('existing-key');
+    should(result).eql(value);
+  });
+
+  it('zcache_get: should return null for a non-existing key', async () => {
+    mockRpcCall(JSON.stringify(null));
+
+    const result = await cache.get('non-existing-key');
+    should(result).eql(null);
+  });
+
+  it('zcache_get: should throw error for non-string keys', async () => {
+    await cache.get(12345).should.be.rejectedWith('key must be a string');
+  });
+
+  it('zcache_set: should set a cache entry when current throughput is within app\'s rate-limit', async () => {
+    const key = 'random-key'
+    const value = {entity:'Zapier', colors: ['Orange', 'black']};
+    mockRpcCall(JSON.stringify(true));
+
+    const result = await cache.set(key, value);
+    should(result).eql(true);
+  });
+
+  it('zcache_set: should not set a cache entry when current throughput is outside app\'s rate-limit', async () => {
```

Comment:
Correct me if I'm wrong, but are we using the previous "set" test case as a basis for this test case, otherwise, I'm having trouble understanding what the difference in behaviour is between the two. I recommend having isolated test cases if the above is true

Reply:
Makes sense. I've combined the two test cases.